odoo/tools/translate.py

```diff
@@ -1069,6 +1070,25 @@ def _babel_extract_terms(self, fname, path, root, extract_method="python", trans
         module, fabsolutepath, _, display_path = self._verified_module_filepaths(fname, path, root)
         if not module:
             return
+        if module != self._po_module:
+            self._po_module = module
+            paths = [('_mo_extra', join(root, 'i18n_extra', self._lang + '.po')),
+                     ('_mo_base_extra', join(root, 'i18n_extra', self._lang.split('_')[0] + '.po')),
+                     ('_mo', join(root, 'i18n', self._lang + '.po')),
+                     ('_mo_base', join(root, 'i18n', self._lang.split('_')[0] + '.po'))] if self._lang else []
+            for attr, po_path in paths:
+                try:
+                    with open(po_path, 'r') as p:
+                        catalog = read_po(p)
+                        buf = io.BytesIO()
+                        write_mo(buf, catalog)
+                        buf.seek(0)
+                        setattr(self, attr, GNUTranslations(fp=buf))
+                except FileNotFoundError:
+                    setattr(self, attr, None)
+                except Exception:
+                    raise Exception
```

Comment:
Does all of this really make sense? We seem to be parsing the POs, converting everything to MOs, immediately loading the in-memory MOs into a catalog, then use that directly.

The entire MO step seems unnecessary.